src/app/api.py

```diff
@@ -1,127 +1,74 @@
-
-            if not debug or TIME_REQUIRED not in meta[key].keys():
-                meta[key][TIME_REQUIRED] = None
-
-            extractor_tags.__setattr__(
-                key,
-                MetadataTags(
-                    values=meta[key][VALUES],
-                    stars=meta[key][STAR_CASE],
-                    time_for_completion=meta[key][TIME_REQUIRED],
-                    explanation=meta[key][EXPLANATION],
-                ),
-            )
-
-    return extractor_tags
+@app.middleware("http")
+async def caching_and_response_time(request: Request, call_next):
+    start_time = time.time()
+    response = await call_next(request)
+    process_time = time.time() - start_time
+    response.headers["X-Process-Time"] = str(round(process_time * 1000, 2) / 1000)
+    return response
 
 
 @app.post(
     "/extract_meta",
     response_model=Output,
     description="The main endpoint for metadata extraction.",
 )
-async def extract_meta(input_data: Input):
-    starting_extraction = get_utc_now()
-
-    allowance = json.loads(input_data.allow_list.json())
-
-    # build a list of extractors that are set to True
-    whitelist = None if input_data.allow_list is None else [k for k, v in allowance.items() if v]
-    bypass_cache = False if input_data.bypass_cache is None else input_data.bypass_cache
-    meta_data: dict = await app.manager.start(
-        message=Message(
-            url=input_data.url,
-            splash_response=input_data.splash_response,
-            whitelist=whitelist,
-            bypass_cache=bypass_cache,
-        )
-    )
-
-    if meta_data:
-        extractor_tags = _convert_dict_to_output_model(meta_data, input_data.debug)
-
-        if MESSAGE_EXCEPTION in meta_data.keys():
-            exception = meta_data[MESSAGE_EXCEPTION]
-        else:
-            exception = ""
-
-    else:
-        extractor_tags = None
-        exception = f"No response from {METADATA_EXTRACTOR}."
-
-    end_time = get_utc_now()
-    out = Output(
-        url=input_data.url,
-        meta=extractor_tags,
-        exception=exception,
-        time_until_complete=end_time - starting_extraction,
-    )
+@cache(
+    expire=24 * 60 * 60 * 28,
+    key=lambda input, request, response: str(input.url) if input.splash_response is None else None,
+    backend=app.cache_backend,
+)
+# request and response arguments are needed for the cache wrapper.
+async def extract_meta(input: Input, request: Request, response: Response):
+    """The extract endpoint"""
+    app.logger.info(f"Received request for {input.url}")
 
-    if exception != "":
-        raise HTTPException(
-            status_code=400,
-            detail={
-                MESSAGE_URL: input_data.url,
-                "meta": meta_data,
-                MESSAGE_EXCEPTION: exception,
-                "time_until_complete": end_time - starting_extraction,
-            },
-        )
+    result = await app.manager.extract(input)
```

Comment:
The state that was in the manager needs to live somewhere - Otherwise we would have to setup the extractors for every request (including downloading some taglists).
Another alternative would be simply have a global variable - I have no real preference for that!

Reply:
Me neither. Probably this indicates, that some major refactoring would help us in the future. Fine for now.

Reply:
Me neither. Probably this indicates, that some major refactoring would help us in the future. Fine for now.

Reply:
i moved them into global scope